Problem with 0.15.3 wheels on CircleCI

I'm using pysam on CircleCI builds and I've hit an issue with the just-released 0.15.3 binary wheel (pysam-0.15.3-cp36-cp36m-manylinux1_x86_64.whl)

I get the following error:
```
  File "/home/circleci/tskit/python/tests/test_vcf.py", line 40, in <module>
    import pysam
  File "/home/circleci/.local/lib/python3.6/site-packages/pysam/__init__.py", line 5, in <module>
    from pysam.libchtslib import *
  File "pysam/libchtslib.pyx", line 325, in init pysam.libchtslib
AttributeError: type object 'pysam.libchtslib.HTSFile' has no attribute '__reduce_cython__'
```

I'm not sure how to debug the problem, but perhaps a good short-term solution would be to delete the linux wheel from PyPI temporarily so users download and build the source packages as before?

I tried building this locally with cibuildwheel but it failed with`
```
   build/temp.linux-x86_64-3.6/htslib/regidx.o: In function `regidx_parse_bed':
  /tmp/pip-req-build-izxocbib/htslib/regidx.c:288: multiple definition of `regidx_parse_bed'
  build/temp.linux-x86_64-3.6/bcftools/regidx.c.pysam.o:/tmp/pip-req-build-izxocbib/bcftools/regidx.c.pysam.c:433: first defined here
  build/temp.linux-x86_64-3.6/htslib/regidx.o: In function `regidx_parse_tab':
  /tmp/pip-req-build-izxocbib/htslib/regidx.c:313: multiple definition of `regidx_parse_tab'
  build/temp.linux-x86_64-3.6/bcftools/regidx.c.pysam.o:/tmp/pip-req-build-izxocbib/bcftools/regidx.c.pysam.c:465: first defined here
  build/temp.linux-x86_64-3.6/htslib/regidx.o: In function `regidx_seq_nregs':
  /tmp/pip-req-build-izxocbib/htslib/regidx.c:66: multiple definition of `regidx_seq_nregs'
  build/temp.linux-x86_64-3.6/bcftools/regidx.c.pysam.o:/tmp/pip-req-build-izxocbib/bcftools/regidx.c.pysam.c:89: first defined here
  build/temp.linux-x86_64-3.6/htslib/regidx.o: In function `regidx_nregs':
  /tmp/pip-req-build-izxocbib/htslib/regidx.c:75: multiple definition of `regidx_nregs'
  build/temp.linux-x86_64-3.6/bcftools/regidx.c.pysam.o:/tmp/pip-req-build-izxocbib/bcftools/regidx.c.pysam.c:98: first defined here
  build/temp.linux-x86_64-3.6/htslib/regidx.o: In function `regidx_seq_names':
  /tmp/pip-req-build-izxocbib/htslib/regidx.c:81: multiple definition of `regidx_seq_names'
  build/temp.linux-x86_64-3.6/bcftools/regidx.c.pysam.o:/tmp/pip-req-build-izxocbib/bcftools/regidx.c.pysam.c:104: first defined here
  build/temp.linux-x86_64-3.6/htslib/regidx.o: In function `regidx_insert':
  /tmp/pip-req-build-izxocbib/htslib/regidx.c:120: multiple definition of `regidx_insert'
  build/temp.linux-x86_64-3.6/bcftools/regidx.c.pysam.o:/tmp/pip-req-build-izxocbib/bcftools/regidx.c.pysam.c:183: first defined here
  build/temp.linux-x86_64-3.6/htslib/regidx.o: In function `regidx_destroy':
  /tmp/pip-req-build-izxocbib/htslib/regidx.c:224: multiple definition of `regidx_destroy'
  build/temp.linux-x86_64-3.6/bcftools/regidx.c.pysam.o:/tmp/pip-req-build-izxocbib/bcftools/regidx.c.pysam.c:278: first defined here
  build/temp.linux-x86_64-3.6/htslib/regidx.o: In function `regidx_init':
  /tmp/pip-req-build-izxocbib/htslib/regidx.c:170: multiple definition of `regidx_init'
  build/temp.linux-x86_64-3.6/bcftools/regidx.c.pysam.o:/tmp/pip-req-build-izxocbib/bcftools/regidx.c.pysam.c:224: first defined here
  build/temp.linux-x86_64-3.6/htslib/regidx.o: In function `regidx_overlap':
  /tmp/pip-req-build-izxocbib/htslib/regidx.c:247: multiple definition of `regidx_overlap'
  build/temp.linux-x86_64-3.6/bcftools/regidx.c.pysam.o:/tmp/pip-req-build-izxocbib/bcftools/regidx.c.pysam.c:368: first defined here
  collect2: error: ld returned 1 exit status
  error: command 'gcc' failed with exit status 1
  ----------------------------------------
  ERROR: Failed building wheel for pysam
  Running setup.py clean for pysam
Failed to build pysam
ERROR: Failed to build one or more wheels
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -4,6 +4,11 @@ os:
 
 language: c
 
+stages:
+  - test
+  - name: deploy
+    # if: tag IS present
+
 env:
   matrix:
     - CONDA_PY=2.7
@@ -14,10 +19,24 @@ env:
     - TWINE_USERNAME=grepall
     - secure: bTbky3Un19NAl62lix8bMLmBv9IGNhFkRXlZH+B253nYub7jwQwPQKum3ct9ea+XHJT5//uM0B8WAF6eyugpNkPQ7+S7SEH5BJuCt30nv6qvGhSO2AffZKeHEDnfW2kqGrivn87TqeomlSBlO742CD/V0wOIUwkTT9tutd+E7FU=
 
-_deploy_common: &deploy_common
-  # re-add `if: tag IS present` when wheels work properly
+_cibw_common: &cibw_common
+  addons: {}
   install:
     - python3 -m pip install cibuildwheel>=1.1.0 twine
+  script:
+    - set -e
+    - cibuildwheel --output-dir dist
+    - twine check dist/*
+    # - twine upload --skip-existing dist/*
+
+_cibw_linux: &cibw_linux
+  stage: deploy
+  os: linux
+  language: python
+  python: '3.5'
+  services:
+    - docker
+  <<: *cibw_common
 
 matrix:
   include:
@@ -25,13 +44,6 @@ matrix:
       os: linux
       language: python
       python: '3.5'
-      services:
-        - docker
-      env:
-        - CIBW_BEFORE_BUILD="yum install -y zlib-devel bzip2-devel xz-devel && python -m pip install -r requirements.txt"
-        - CIBW_ENVIRONMENT='HTSLIB_CONFIGURE_OPTIONS="--disable-libcurl"'
-        - CIBW_REPAIR_WHEEL_COMMAND_LINUX='auditwheel repair -L . -w {dest_dir} {wheel}'
-        - CIBW_TEST_COMMAND='python -c "import pysam"'
       addons:
         apt:
           packages:
@@ -39,15 +51,28 @@ matrix:
             - g++
             - libcurl4-openssl-dev  # for libcurl support in sdist
             - libssl-dev  # for s3 support in sdist
-      <<: *deploy_common
+      install:
+        - python3 -m pip install Cython twine
       script:
         - set -e
-        - cibuildwheel --output-dir dist
-        - python3 -m pip install Cython
         - python3 setup.py build_ext --inplace
         - python3 setup.py sdist
         - twine check dist/*
-#        - twine upload --skip-existing dist/*
+        # - twine upload --skip-existing dist/*
+    - <<: *cibw_linux
+      env:
+        - CIBW_BUILD="*_x86_64"
+        - CIBW_BEFORE_BUILD="yum install -y zlib-devel bzip2-devel xz-devel && python -m pip install -r requirements.txt"
+        - CIBW_ENVIRONMENT='HTSLIB_CONFIGURE_OPTIONS="--disable-libcurl"'
+        - CIBW_REPAIR_WHEEL_COMMAND_LINUX='auditwheel repair -L . -w {dest_dir} {wheel}'
+        - CIBW_TEST_COMMAND='python -c "import pysam"'
+    - <<: *cibw_linux
+      env:
+        - CIBW_BUILD="*_i686"
+        - CIBW_BEFORE_BUILD="yum install -y zlib-devel bzip2-devel xz-devel && python -m pip install -r requirements.txt"
+        - CIBW_ENVIRONMENT='HTSLIB_CONFIGURE_OPTIONS="--disable-libcurl"'
+        - CIBW_REPAIR_WHEEL_COMMAND_LINUX='auditwheel repair -L . -w {dest_dir} {wheel}'
+        - CIBW_TEST_COMMAND='python -c "import pysam"'
     - stage: deploy
       os: osx
       language: generic
@@ -55,13 +80,7 @@ matrix:
         - CIBW_BEFORE_BUILD="python -m pip install -r requirements.txt"
         - CIBW_ENVIRONMENT='HTSLIB_CONFIGURE_OPTIONS="--disable-libcurl"'
         - CIBW_TEST_COMMAND='python -c "import pysam"'
-      addons: {}
-      <<: *deploy_common
-      script:
-        - set -e
-        - cibuildwheel --output-dir dist
-        - twine check dist/*
-#       - twine upload --skip-existing dist/*
+      <<: *cibw_common
 
 addons:
   apt:

--- a/.travis.yml
+++ b/.travis.yml
@@ -15,9 +15,9 @@ env:
     - secure: bTbky3Un19NAl62lix8bMLmBv9IGNhFkRXlZH+B253nYub7jwQwPQKum3ct9ea+XHJT5//uM0B8WAF6eyugpNkPQ7+S7SEH5BJuCt30nv6qvGhSO2AffZKeHEDnfW2kqGrivn87TqeomlSBlO742CD/V0wOIUwkTT9tutd+E7FU=
 
 _deploy_common: &deploy_common
-  if: branch = master AND type = push AND fork = false
+  # re-add `if: tag IS present` when wheels work properly
   install:
-    - python3 -m pip install cibuildwheel twine
+    - python3 -m pip install cibuildwheel>=1.1.0 twine
 
 matrix:
   include:
@@ -28,8 +28,9 @@ matrix:
       services:
         - docker
       env:
-        - CIBW_BEFORE_BUILD="yum install -y zlib-devel bzip2-devel xz-devel && pip install -r requirements.txt"
+        - CIBW_BEFORE_BUILD="yum install -y zlib-devel bzip2-devel xz-devel && python -m pip install -r requirements.txt"
         - CIBW_ENVIRONMENT='HTSLIB_CONFIGURE_OPTIONS="--disable-libcurl"'
+        - CIBW_REPAIR_WHEEL_COMMAND_LINUX='auditwheel repair -L . -w {dest_dir} {wheel}'
         - CIBW_TEST_COMMAND='python -c "import pysam"'
       addons:
         apt:
@@ -51,7 +52,7 @@ matrix:
       os: osx
       language: generic
       env:
-        - CIBW_BEFORE_BUILD="pip install -r requirements.txt"
+        - CIBW_BEFORE_BUILD="python -m pip install -r requirements.txt"
         - CIBW_ENVIRONMENT='HTSLIB_CONFIGURE_OPTIONS="--disable-libcurl"'
         - CIBW_TEST_COMMAND='python -c "import pysam"'
       addons: {}

